### PR TITLE
[AC-xx: cicd]  EKS - Add dockerfile, jenkinfile and deployment file 

### PR DESCRIPTION
**Pull request by mable-pmyip**
### Provide description as to what you have done in this pr.
Add dockerfile, jenkinfile and deployment file 
### Provide screenshots or videos that the function is working.
image

### Provide screenshots of no warning, error in console.
image

### CODE REVIEW

File: CI-CD/Jenkinsfile.Mable [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		stage('Test'){
+		  steps {
+			sh 'echo unit testing'
+		  }
+		}
+		
+		stage('Build App Image') {
+		   steps {
+            script {
+				dockerImage = docker.build( appRegistry + ":$BUILD_NUMBER", "-f Dockerfile.Alex .")
```
Comment (mable-pmyip) [inline CI-CD/Jenkinsfile.Mable]: my dockerfile was working last week but not sure why doesn't work recently, stole from Alex for now :P

Comment (tardigrade425) [inline CI-CD/Jenkinsfile.Mable]: 禁止引用他人名称文件

**Review by tardigrade425 — CHANGES_REQUESTED** [review]

File: Dockerfile.Mable [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+RUN yarn build
+FROM node:14
+COPY --from=builder /app/node_modules ./node_modules
+COPY --from=builder /app/package*.json ./
+COPY --from=builder /app/tsconfig*.json ./
+COPY --from=builder /app/dist ./dist
+
+EXPOSE 3100
+
+# for primas
```
Comment (tardigrade425) [inline Dockerfile.Mable]: Secrets不允许明文出现在Repo中

File: deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: comosus-server-jenkins 
+  labels:
+    app: comosus-server
```
Comment (tardigrade425) [inline deployment.yaml:16]: 这是deploy to个人的还是项目的？

**Review by tardigrade425 — CHANGES_REQUESTED** [review]

File: deployment.yaml (hunk 3 above)
Comment (mable-pmyip) [reply deployment.yaml:16]: 个人的

File: CI-CD/Jenkinsfile.Mable (hunk 1 above)
Comment (mable-pmyip) [reply CI-CD/Jenkinsfile.Mable]: updated to my dockerfile

File: Dockerfile.Mable (hunk 2 above)
Comment (mable-pmyip) [reply Dockerfile.Mable]: deleted secret

**Comment by mable-pmyip** [comment]
@tardigrade425 Hi William, sorry for the delay, I just fixed the issues you mentioned, can you pls review the PR again. Thanks.

File: deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: app-secret
+type: Opaque
+data:
+  DATABASE_URL: bW9uZ29kYitzcnY6Ly9hZG1pbjphZG1pbkBjbHVzdGVyMC5vcG1uZS5tb25nb2RiLm5ldC9hLWNvbW9zdXM/cmV0cnlXcml0ZXM9dHJ1ZSZ3PW1ham9yaXR5
```
Comment (mable-pmyip) [inline deployment.yaml:8]: Passing the encoded database_url here in the secret def file.

File: CI-CD/Jenkinsfile.Mable [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+		registryCredential = 'ecr:us-east-1:awscreds'
+		appRegistry = "947036392937.dkr.ecr.us-east-1.amazonaws.com/comosus-server-img"
+		comosusClientRegistry = "https://947036392937.dkr.ecr.us-east-1.amazonaws.com/"
+	}
+	
+	tools { nodejs "node" }
+	
+	stages {		    
+	        stage('Clear space'){
+		  steps {
```
Comment (tardigrade425) [inline CI-CD/Jenkinsfile.Mable]: 缩进，对齐

Comment (tardigrade425) [reply CI-CD/Jenkinsfile.Mable]: 下同

File: Dockerfile.Mable [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+# Production image, copy all the files and run next
+FROM node:alpine AS runner
+WORKDIR /app
+ENV NODE_ENV production
+COPY --from=builder /app/dist ./dist
+COPY --from=builder /app/node_modules ./node_modules
+COPY --from=builder /app/package.json ./package.json
+COPY --from=builder /app/tsconfig.build.json ./tsconfig.build.json
+
+ENV PORT 3100
```
Comment (tardigrade425) [inline Dockerfile.Mable:22]: 这里的环境变量和deployment中的有什么关系？

Comment (mable-pmyip) [reply Dockerfile.Mable:22]: Not quite sure, I am just using the DATABASE_URL at the moment (in deployment.yaml file)
but may be required when using terraform to deploy. 
@AlexXin14 Can you please confirm that?

**Comment by NorrisWu0** [comment]
Nope. That error means either db is not connected, or your db schema has not been updated. Try running `yarn prisma db push` before building the backend code.

**Comment by mable-pmyip** [comment]
> Nope. That error means either db is not connected, or your db schema has not been updated. Try running `yarn prisma db push` before building the backend code.

Thanks, will have a try tonight

**Comment by mable-pmyip** [comment]
> Nope. That error means either db is not connected, or your db schema has not been updated. Try running `yarn prisma db push` before building the backend code.

The  `yarn prisma db push` doesn't work for me, but it's working fine when I created a new  account. 
Seems like because the old account doesn't have the createdAt field as I created it a few weeks ago. Anyway, it's all working now. Thanks! 
image

File: CI-CD/Jenkinsfile.Mable [hunk 7]
```diff
@@ -0,0 +1,69 @@
+pipeline {
+    agent any
+
+    tools {
+        nodejs 'node'
+    }
+
+    environment {
+        registryCredential = 'ecr:us-east-1:awscreds'
```
Comment (tardigrade425) [inline CI-CD/Jenkinsfile.Mable:9]: 对齐

File: CI-CD/Jenkinsfile.Mable [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        stage('Build App Image') {
+            steps {
+                script {
+                    dockerImage = docker.build( appRegistry + ":$BUILD_NUMBER", "-f Dockerfile.Mable .")
+		}
+            }
+        }
+
+        stage('Upload App Image') {
+	    steps{
```
Comment (tardigrade425) [inline CI-CD/Jenkinsfile.Mable:49]: 缩进

Comment (tardigrade425) [reply CI-CD/Jenkinsfile.Mable:49]: 下面还有

File: Dockerfile.Mable [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+ENV NODE_ENV staging
+ENV URL_META_AUTH_STRING gmail
+ENV DATABASE_URL mongodb
```
Comment (tardigrade425) [inline Dockerfile.Mable:25]: k8s secrets中的变量，这里是否还有必要？

File: CI-CD/Jenkinsfile.Mable [hunk 10]
```diff
@@ -0,0 +1,69 @@
+pipeline {
+    agent any
+
+    tools {
+        nodejs 'node'
```
Comment (tardigrade425) [inline CI-CD/Jenkinsfile.Mable:5]: node版本号

**Review by tardigrade425 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**